dashboard/app/models/script.rb

```diff
@@ -330,7 +330,8 @@ def self.flappy_unit
   end
 
   def self.maker_units(user)
-    return_units = @@maker_units ||= visible_units.select(&:is_maker_unit?)
+    # only units in CSD should be included in the maker units
+    return_units = @@maker_units ||= visible_units.select(&:is_maker_unit?).select {|u| u.course_version&.course_offering&.key == 'csd'}
```

Comment:
@dmcavoy This isn't my lane, but this comment stuck out to me.

For the CSD Modularity pilot, Jess is making "Curriculum Investigation" scripts. One of them is Unit 6 (https://levelbuilder-studio.code.org/s/vpl-csd6-ci-pilot-2022). When she was putting this together, I told her to check "Is a maker unit" because... well, it made sense and it borrows levels from CSD Unit 6.

Will that mess anything up with this logic? Or maybe I don't actually have a question here and instead I'm just flagging that there's at least one script not in CSD that we would expect to work in the Maker App (the script I linked above, part of a PL course).

Reply:
Thanks for taking a look Dan. 

The "Maker Unit" setting is only used to determine which unit will show up on the maker home page. We already have another place where we check to make sure that the course is CSD:

https://github.com/code-dot-org/code-dot-org/blob/ec044998ad575765f02d185612d9eb7a02002e1e/dashboard/app/controllers/maker_controller.rb#L29

As part of this PR I also went through and remove the maker setting from all the units which are not specifically in CSD (pilots, plc, etc). So vpl-csd6-ci-pilot-2022 no longer will have `is_maker_unit` set. It should not impact any function of using the maker tools.

Reply:
I added a comment on the method to hopefully help clarify this going forward

Reply:
Also this new check would make it so if `is_maker_unit` got checked for a unit outside CSD it would not cause any issues. However it would be good for us to just not even show the setting unless the course offering is CSD.

Reply:
And now I have made it so you will only see this setting if a unit is part of the CSD course offering.

Reply:
Thanks Dani, and sorry for the detour =/. I think I've learned that I never really understood what the "is maker unit" checkbox does on the script edit page, so thanks for your work addressing that too =)